packages/loot-core/src/server/budget/goaltemplates.ts

```diff
@@ -529,7 +529,10 @@ async function applyCategoryTemplate(
               }
               next_date = addDays(next_date, 1);
               next_date_string = getNextDate(dateCond, next_date);
-              next_date = new Date(next_date_string);
+              let new_next_date = new Date(next_date_string);
+              new_next_date = addDays(new_next_date, 1);
+              if (next_date === new_next_date) break;
```

Comment:
I don’t think this will works Ince the dates will always be different objects even if they refer to the same date.

Reply:
Maybe it’s an issue with time zones? Midnight UTC on June 30 may still be June 29 in another time zone

Reply:
You are correct.  If I append 'T11:00' to next_date_string, it interprets correctly.  I've never dealt with dates before, any guidance on the best way to deal with this?